smartcontractkit/external-adapters-js
# Change DP error status code surfacing

## Description

A common occurrence with EAs is that the Data Provider sources fail, and we currently surface their status code as the response from the EA itself. Since EAs are not a completely clean proxy by design, the end result is that it's not possible to discern in metrics whether the failing point is the EA or the DP.

This PR changes the behavior of the core bootstrap so that a failing request to a DP results in a `200` response from the EA, providing a new field `providerStatusCode` to give complete information to the requester. The reasoning behind the new field was to avoid the confusing perspective of a single `statusCode` in the response, that used to match the HTTP status but would no longer do so, as well as leaving the existing behavior intact. It's true that the `statusCode` field will remain somewhat redundant, which is a change that could be made in the future to remove it altogether.

**Note**: the first commit includes the changes to the core package, the other two are fixes for all tests that failed from those changes.

## Changes

-  Change the behavior of the core requester to surface response error status code as a new field, with main status code changing to 200



## Steps to Test

1. Run any adapter of choice
2. Make sure that whatever input you provide will cause an error in the API (not in the adapter)
3. The response should have a `statusCode` of 200, and a `providerStatusCode` of whatever the original API failure code was

## Quality Assurance
